### ISSUE TITLE
Add SliverSizedBox (perhaps SliverConstrainedBox, SliverLimitedBox, etc?)

### ISSUE THREAD

**Issue by Reprevise** [flutter/flutter#125797]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Use case

I personally like to add space between widgets via a `SizedBox` in between. I could alternatively use `SliverPadding` with a `null` child but that doesn't feel "correct." There is a package called `gap` which would work, however it is unable to take a child to size. Right now I've created my own `SliverSizedBox` widget by just wrapping a `SizedBox` in a `SliverToBoxAdapter` but I feel like a proper sliver widget is necessary in the framework.

### Proposal

Add a new sliver widget named `SliverSizedBox` that is similar to the normal box `SizedBox` however instead of it taking a nullable box `child`, it will take a sliver `child`.

Perhaps other widgets can be added too like `SliverConstrainedBox` and `SliverLimitedBox`.

**Comment by Piinks** [comment]
Would this sliver take sliver children or box children?

**Comment by Reprevise** [comment]
Hey @Piinks,

I think `SliverSizedBox` would take a nullable sliver child.

**Comment by Piinks** [comment]
Interesting! Thanks for the detail. Have you seen `SliverConstrainedCrossAxis`? It landed recently in https://github.com/flutter/flutter/pull/125239 

Can you help me understand the expectation for the main axis? If a SliverList for example had 400 pixels worth of children in the main axis, but it were wrapped in a constraining sliver in the main axis of say 200 pixels, would the SliverList then just get cut off?

**Comment by Reprevise** [comment]
The expectation would be that the list view itself would be restricted to 200px but the inner extent of the list view would be the 400px, if I'm understanding correctly. That's how I would assume it'd function.

**Comment by Piinks** [comment]
That's awesome! Glad to hear it!

>  (so you can scroll its contents)

Ah, this would be tricky. Slivers don't actually scroll, they are provided information about the position and layout accordingly. They are viewed through a viewport (like a window) which controls what part of them you see (based on the offset). So having a sliver scroll inside of a sliver would be something new entirely. 

Although, this does sound maybe like a parallax effect? So scrolling on the scrollable (which contains a sliver constrainer that contains a sliver list) would scroll both the whole scroll view and the inner sliver list? Not sure, but I appreciate you sharing your thoughts!

**Comment by Tienisto** [comment]
This would be a great improvement. I always write

```dart
const SliverToBoxAdapter(
  child: SizedBox(
    height: 20,
  ),
),
```

which is kind of verbose

**Comment by Velkamhell97** [comment]
Still waiting for this 🥲

**Comment by mateusfccp-cos** [comment]
It's interesting that we still haven't this as of 2026.

I would be willing to open a PR with this if there's any interest from the Flutter team.

I would rather call it `SizedSliver`, as it is consistent with `SizedBox`. `SizedSliver` would be the `RenderSliver` equivalent of `SizedBox`. The behavior would be the same as `SizedBox`, but applied to one-dimensional objects (slivers).

That would be basically the same as we currently do with `SliverToBoxAdapter` + `SizedBox`.

**Comment by mateusfccp** [comment]
@Piinks I agree that we can achieve this with composition. However, for very common use cases it's always good to have a simplified API. That's why, for instance, we have `Container` even though we can compose `DecoratedBox`, `SizedBox`, `Align`, etc.
